### ISSUE TITLE
Out of memory using libgpuarray and gpuarray.preallocate

### ISSUE THREAD

**Issue by jessejohns** [Theano/Theano#4993]
When I am using libgpuarray and have set gpuarray.preallocate = 1, I receive the following error:

```
RuntimeError: ('The following error happened while compiling the node', GpuDnnConvGradW{algo='none', inplace=True}(GpuContiguous.0, GpuContiguous.0, GpuContiguous.0, GpuDnnConvDesc{border_mode='half', subsample=(1, 1), conv_mode='conv', precision='float32'}.0, Constant{1.0}, Constant{1.0}), '\n', 'could not create cuDNN handle: CUDNN_STATUS_INTERNAL_ERROR')                         
Segmentation fault (core dumped)
```

When global.device = gpu and lib.cnmem = 1, the model runs just fine.  To fix this, I set gpuarray.preallocate = 0.8.

**Comment by gaebor** [comment]
I have tried the new gpu backend (libgpuarray) with preallocation.

You cannot allocate 100% of the GPU memory, it is capped at 95% by the library.
However, I managed to do the preallocation, I tried with two different sessions with 0.5 preallocation in both of them. The second session failed, because I didn't have enough memory (as expected).

See your memory usage with `nvidia-smi`, it may happen that you have several hundred MB of memory allocated by trailing processes, which makes your 95% preallocation impossible.
